### DISCUSSION TITLE
Unable to make SandDance visual interact with other Power BI plots

### DISCUSSION THREAD

**Discussion by sahitkogs** [microsoft/SandDance#596] [Q&A] [answered]
Hi, I am trying to make SandDance visual interact with other Power BI visuals and table visual. 
Say, in a density plot in SandDance, when I click on a particular bar, it does not filter other plots or table visual. 
But when I to do the opposite, everything works as expected. Meaning, when I click on a bar in "stacked bar chart", the visual on SandDance also gets filtered.

Below is an example of my problem.
In the below example, even though product "Amarilla" is selected is SandDance, it is not filtered in bar plot or table visual.
![image](https://user-images.githubusercontent.com/6560322/213756479-a97995f4-73be-433b-bde9-922272cfa018.png)

but in the below visual, when I selected the same product on bar plot, the filtering is reflected in all the visuals
![image](https://user-images.githubusercontent.com/6560322/213757702-8481fa28-bff8-44a0-b8b4-7ad621dcf007.png)

Can you please let me know if I can correct this and how to?

**Comment by danmarshall** [comment]
Hello, in this [example pbix](https://microsoft.github.io/SandDance/dist/powerbi/v4/SandDance201929976D117A654D0BAB8E96507442D80B.4.1.2.0.pbix), clicking on an item will filter the table for me. Can you try and see if it works for you?

**Reply by sahitkogs** [reply]
Hi Dan, Thanks for the pbix file. The interaction works as expected. Can you let me if it has same pbiviz file that is available to download for us? Are there any other special changes that you made to make this work?

**Reply by sahitkogs** [reply]
I have created a new page in the same file and tried to recreate the same visuals you did. But surprisingly, the interaction between the SandDance visual and other native plots doesn't work as expected. Can you let me know what I am missing here?
[SandDance201929976D117A654D0BAB8E96507442D80B.4.1.2.0_v2.zip](https://github.com/microsoft/SandDance/files/10471722/SandDance201929976D117A654D0BAB8E96507442D80B.4.1.2.0_v2.zip)

**Reply by danmarshall** [reply]
I've opened your file and my selections showed the same results in the table & stacked bar chart:
<img width="738" alt="image" src="https://user-images.githubusercontent.com/11507384/214125326-fd50b543-dea8-41a3-b343-fd02331c01db.png">
Is this the same behavior you get?

**Reply by sahitkogs** [reply]
Hi Dan, In the "Page 1" I see the interaction working fine. But in the "Page1_Sahit" tab (which I created from scratch) I do not see the interaction working. I'm not sure why this is happening.

I also tried doing below steps:
I extracted the pbiviz file of SandDance 4.1.2 version from the pbix file you shared. In a new Power BI window, I activated the developer option and imported the 4.1.2 version of SandDance. Imported my own dataset and tested if the interaction is working. But no success. Interaction doesn't work. Can you check if you can achieve the interaction in the attached file?
[sample.zip](https://github.com/microsoft/SandDance/files/10486500/sample.zip)

**Reply by danmarshall** [reply]
Thank you for sample, it was helpful to see your repro. The key thing you were missing was to select `Don't summarize` for the fields you are providing to SandDance:

<img width="276" alt="image" src="https://user-images.githubusercontent.com/11507384/214682895-d5b5a709-7e32-4367-ae6f-f0a98b7c8c97.png">

SandDance is different in this regard among typical Power BI visuals - it needs unique, identifiable data because it does the aggregation and summarization visually with each data row becoming a "pixel" unit in a chart.

Apologies that this key was not easily discoverable. If you have any suggestions on how to improve documentation, please let me know (or better yet, submit a PR 😊)

**Comment by danmarshall** [comment, marked as answer]
Select `Don't summarize` for the fields you are providing to SandDance:

<img width="276" alt="image" src="https://user-images.githubusercontent.com/11507384/214682895-d5b5a709-7e32-4367-ae6f-f0a98b7c8c97.png">

SandDance is different in this regard among typical Power BI visuals - it needs unique, identifiable data because it does the aggregation and summarization visually with each data row becoming a "pixel" unit in a chart.
